### DISCUSSION TITLE
Question about GPT 4 model

### DISCUSSION THREAD

**Discussion by thomclae33** [khoj-ai/khoj#187] [Q&A] [answered]
Is there a way to make the gpt-4 model work with Khoj, or does this need a lot of changes?

**Comment by debanjum** [comment, marked as answer]
What's your use-case for GPT4 with Khoj?

From my understanding, generating embedding via GPT4 isn't supported by OpenAI yet. So GPT4 can't be used for Search (with Khoj or otherwise).

But using GPT4 models for Khoj Chat shouldn't be much work as Khoj was just moved to start using ChatGPT for Chat. Both GPT4 and ChatGPT use the new ChatML format to provide context.

So I'll just need to make the Chat model to use configurable via `khoj.yml`.

**Reply by thomclae33** [reply]
Thank you for the clear info! One use case / hope was to use more data from my Obsidian vault for prompts and make more meaningful connections thanks to the better reasoning of GPT4, and ask these more complex summaries through Khoj (or even ask it complex questions via chat).

**Reply by debanjum** [reply]
I see. Yeah, being able to pass in more notes as context to GPT4 should improve the answers that Khoj Chat can provide. 

Do try the latest Khoj Chat, it's still a work in progress, but see if it helps you find any meaningful connections or provide other forms of assistance. It can currently be installed as a pre-release with `pip install --upgrade --pre khoj-assistant`. See [this section ](https://github.com/debanjum/khoj#Khoj-chat )for other setup/usage details for Khoj Chat.
